Master-Game/World/Map.cs

```diff
@@ -39,5 +53,91 @@ public void ResetTiles()
                 }
             }
         }
+
+        public MapData LoadJson()
+        {
+            string fileName = "Jay01.json";
+            string filePath = Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory())) + "\\Assets\\Maps\\" + fileName;
+            FileInfo f = new FileInfo(fileName);
+            string fullName = f.Name;
+            using (StreamReader reader = new StreamReader(filePath))
+            {
+                string myJson = reader.ReadToEnd();
+                MapData mapData = JsonConvert.DeserializeObject<MapData>(myJson);
```

Comment:
Before you deserialize check if the string is empty.

Reply:
No need to store the data in a local variable mapData and then return mapData. Just return it.

Basically return JsonConvert.DeserializeObject<MapData>(myJson);